packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -1420,6 +1395,92 @@ export default class Meeting extends StatelessWebexPlugin {
     }
   }
 
+  /**
+   * Refreshes the meeting info permission token (it's required for joining meetings)
+   *
+   * @returns {Promise}
+   */
+  public async refreshPermissionToken(): Promise<void> {
+    if (!this.meetingInfo?.permissionToken) {
+      return;
+    }
+
+    const isStartingSpaceInstantV2Meeting =
+      this.destinationType === _CONVERSATION_URL_ &&
+      // @ts-ignore - config coming from registerPlugin
+      this.config.experimental.enableAdhocMeetings &&
+      // @ts-ignore
+      this.webex.meetings.preferredWebexSite;
+
+    const destination = isStartingSpaceInstantV2Meeting
+      ? this.meetingInfo.meetingJoinUrl
+      : this.destination;
+    const destinationType = isStartingSpaceInstantV2Meeting ? _MEETING_LINK_ : this.destinationType;
+
+    await this.fetchMeetingInfoInternal({
```

Comment:
question: the part of catching the `captcha` and `password` errors is not in the internal fetchMeetingInfo function. We want to throw these errors when we refreshPermissionToken no? (for example the password changes while a user waits a long time in interstitial). It seems the captcha and password errors are still getting thrown in the call stack below, but then i am wondering why we had these handlers in line 1457 originally?

Reply:
you mean the checks that we do in fetchMeetingInfo() before we call `this.attrs.meetingInfoProvider.fetchMeetingInfo()`? These are not possible to happen if we already have the permissionToken  (checked in line 1404). If we have the token, it means we must have passed correct password and captcha or they were not required. If password changes or for whatever reason wbxappapi decides we need to show a captcha when refreshing the token, the errors will be thrown by the call to `this.attrs.meetingInfoProvider.fetchMeetingInfo()` and are caught inside fetchMeetingInfoInternal() (line 1339)